### PR TITLE
23-徐晔-hw3

### PR DESCRIPTION
**Pull request by yexu15**
merge lesson 3 from guigulive

### CODE REVIEW

File: Lesson3/assignment/yours.sol [hunk 1]
```diff
@@ -1 +1,95 @@
-/*作业请提交在这个目录下*/
+pragma solidity ^0.4.14;
+
+import './SafeMath.sol';
```
Comment (jonnyhsy) [inline Lesson3/assignment/yours.sol]: import当前目录的文件是可以，但是需要附上这两个文件。或者更简单的做法是：
import "github.com/OpenZeppelin/zeppelin-solidity/contracts/math/SafeMath.sol";
import "github.com/OpenZeppelin/zeppelin-solidity/contracts/ownership/Ownable.sol";
remix会自动fetch这两个库文件。

File: Lesson3/assignment/yours.sol [hunk 2]
```diff
@@ -1 +1,95 @@
...
+    modifier employeeNotExist (address employeeId) {
+        var employee = employees[employeeId];
+        assert(employee.id == 0x0);
+        _;
+    }
+    
+    // 内部function, 用来解决避免重复写code
+    // 此处要把内部函数设为private, 因为用到了在Payroll()中自定义的Employee
+    function _partialPaid(Employee employee) private {
+            uint payment = employee.salary * (now - employee.lastPayday) / payDuration;
```
Comment (jonnyhsy) [inline Lesson3/assignment/yours.sol]: 数学运算需要使用SafeMath库来防止溢出，
```
contract XXX {
  using SafeMath for uint;
  
  function xxx() {
     ...
     uint pay = salary.mul(now.sub(lastPayday)).div(payDuration);
     ...
  }
  ...
}
```
其余几处同理。
文档：https://solidity.readthedocs.io/en/develop/contracts.html#using-for

**Comment by yexu15** [comment]
Thanks @jonnyhsy , please review again. 

One question: I defined modifiers employeeNotExist and employeeExist, but I am wondering if I can only define employeeExist, so that I can use it to modify function addEmployee() ?

**Comment by jonnyhsy** [comment]
It's probably undoable, and the compile emits errors. If it's used only once it might not be necessary to define that into a modifier..

**Pull request merged**